core/src/main/java/ai/timefold/solver/core/impl/score/stream/bavet/BavetConstraintSessionFactory.java

```diff
@@ -81,18 +98,32 @@ public BavetConstraintSession<Score_> buildSession(Solution_ workingSolution, bo
                 castConstraint.collectActiveConstraintStreams(constraintStreamSet);
                 constraintWeightMap.put(constraint, constraintWeight);
             } else {
```

Comment:
```suggestion
} else if (constraintWeightLoggingEnabled) {
```

Reply:
Normally I would apply this change, but considering that this PR is split in two commits and I won't be squashing them, I decided to skip this one. It's mostly an aesthetic change anyway.